lib/client.js

```diff
@@ -248,13 +273,27 @@ class ServiceClient {
 
         return new Promise((resolve, reject) => this.breaker.run(
```

Comment:
It looks like my comment got lost somehow, but here I'm a little worried that you will call `failure` callback multiple times inside a single `breaker.run` call, which doesn't seem logically right and also might cause issues. Can we change that so each retry is also a `breaker.run`?

Reply:
Sure, I'll look into restructuring it this afternoon, any thoughts on how we can provide information on the number of retries performed to the user of perron? I was thinking of possibly adding a value to the response/error object

Reply:
It might be OK, but I'm not sure so far what is the best way to provide information about each stage of the request, would be open to any ideas, as now error / info reporting is not really great in perron at the moment

Reply:
After discussing with @shuhei I will add a callback function like ```onRetry``` that the consumer of perron can define in order to track the retry action how they like